SmartDeviceLink/SDLTextAndGraphicManager.m

```diff
@@ -148,41 +154,47 @@ - (void)sdl_updateWithCompletionHandler:(nullable SDLTextAndGraphicUpdateComplet
     fullShow = [self sdl_assembleShowImages:fullShow];
 
     self.inProgressHandler = handler;
+
     __weak typeof(self)weakSelf = self;
     if (!([self sdl_shouldUpdatePrimaryImage] || [self sdl_shouldUpdateSecondaryImage])) {
-        SDLLogV(@"No images to send, only sending text");
-        // If there are no images to update, just send the text update
+        SDLLogV(@"No images to send, sending text");
+        // If there are no images to update, just send the text
         self.inProgressUpdate = [self sdl_extractTextFromShow:fullShow];
-    } else if ([self sdl_uploadedArtworkOrDoesntExist:self.primaryGraphic] && [self sdl_uploadedArtworkOrDoesntExist:self.secondaryGraphic]) {
+    } else if ([self sdl_artworkAlreadyUploadedOrNonexistent:self.primaryGraphic] && [self sdl_artworkAlreadyUploadedOrNonexistent:self.secondaryGraphic]) {
         SDLLogV(@"Images already uploaded, sending full update");
         // The files to be updated are already uploaded, send the full show immediately
         self.inProgressUpdate = fullShow;
     } else {
-        SDLLogV(@"Images need to be uploaded, sending text and uploading images");
-        // We need to upload or queue the upload of the images
-        // Send the text immediately
-        self.inProgressUpdate = [self sdl_extractTextFromShow:fullShow];
-        // Start uploading the images
-        __block SDLShow *thisUpdate = fullShow;
-        [self sdl_uploadImagesWithCompletionHandler:^(NSError * _Nonnull error) {
-            __strong typeof(weakSelf) strongSelf = weakSelf;
+        SDLLogV(@"Uploading images...");
 
-            if (error != nil) {
-                SDLLogE(@"Error uploading text and graphic image: %@", error);
-            }
-
-            // Check if queued image update still matches our images (there could have been a new Show in the meantime) and send a new update if it does. Since the images will already be on the head unit, the whole show will be sent
-            // TODO: Send delete if it doesn't?
-            if ([strongSelf sdl_showImages:thisUpdate isEqualToShowImages:strongSelf.queuedImageUpdate]) {
-                SDLLogV(@"Queued image update matches the images we need, sending update");
-                [strongSelf sdl_updateWithCompletionHandler:strongSelf.inProgressHandler];
-            } else {
-                SDLLogV(@"Queued image update does not match the images we need, skipping update");
-            }
-        }];
-        
-        // When the images are done uploading, send another show with the images
-        self.queuedImageUpdate = fullShow;
+        if ([self sdl_artworksFinishedUploading:self.primaryGraphic secondaryGraphic:self.secondaryGraphic maxRetryCount:SDLMaxArtworkUploadRetryAttempts]) {
```

Comment:
I don't think this will ever be true? The primary way it _could_ be true is covered by an `else if` above, in which case the full show is sent. Or, if the artworks are beyond the allowable retry limits and therefore aren't available, but in _that_ case, we're trying to show those artworks, which makes no sense.

Reply:
Fixed